src/argus/auth/views.py

```diff
@@ -69,3 +69,16 @@ def get(self, request, *args, **kwargs):
         names = get_psa_authentication_names()
         data = {name: reverse("social:begin", kwargs={"backend": name}, request=request) for name in names}
         return Response(data)
+
+
+class RefreshTokenView(ObtainAuthToken):
+    serializer_class = RefreshTokenSerializer
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+        user = serializer.validated_data["user"]
+        key = serializer.validated_data["old_token"]
+        Token.objects.get(key=key).delete()
+        token = Token.objects.create(user=user)
+        return Response({"token": token.key})
```

Comment:
Needs to be atomic, or you risk deleting a token and not receiving the new one (network goes down! power goes out! absurd lag!) You could decorate the entire method, or just the three last lines, and after the last line raise an error

Reply:
As it is now, if you never receive the new token (lines 83-84) and retry the endpoint, you'll get a DoesNotExist on line 82.

Reply:
I cannot see that this was resolved? @hmpf's comment makes perfect sense; lines 82-84 really need to be wrapped in a transactional decorator, something like
```suggestion
    def post(self, request, *args, **kwargs):
        serializer = self.serializer_class(data=request.data, context={"request": request})
        serializer.is_valid(raise_exception=True)
        user = serializer.validated_data["user"]
        key = serializer.validated_data["old_token"]
        with transaction.atomic():
            Token.objects.get(key=key).delete()
            token = Token.objects.create(user=user)
            return Response({"token": token.key})
```

alternately, the whole method could be decorated with `@transaction.atomic`. See https://docs.djangoproject.com/en/3.0/topics/db/transactions/ for more details.

Reply:
![image](https://user-images.githubusercontent.com/31960753/173063712-ab75db3e-6dc4-4c73-85db-2ce220f0126d.png)
its already wrapped in @transaction.atomic as far as i can see